iroh-sync/src/store/fs/migrate_v1_v2.rs

```diff
@@ -0,0 +1,140 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+use redb::{MultimapTableHandle, TableHandle};
+use redb_v1::{ReadableMultimapTable, ReadableTable};
+use tempfile::NamedTempFile;
+use tracing::info;
+
+macro_rules! migrate_table {
+    ($rtx:expr, $wtx:expr, $old:expr, $new:expr) => {{
+        let old_table = $rtx.open_table($old)?;
+        let mut new_table = $wtx.open_table($new)?;
+        let name = $new.name();
+        let len = old_table.len()?;
+        info!("migrate {name} ({len} rows)..");
+        let ind = (len as usize / 1000) + 1;
+        for (i, entry) in old_table.iter()?.enumerate() {
+            let (key, value) = entry?;
+            let key = key.value();
+            let value = value.value();
+            if i > 0 && i % 1000 == 0 {
+                info!("    {name} {i:>ind$}/{len}");
+            }
+            new_table.insert(key, value)?;
+        }
+        info!("migrate {name} done");
+    }};
+}
+
+macro_rules! migrate_multimap_table {
+    ($rtx:expr, $wtx:expr, $old:expr, $new:expr) => {{
+        let old_table = $rtx.open_multimap_table($old)?;
+        let mut new_table = $wtx.open_multimap_table($new)?;
+        let name = $new.name();
+        let len = old_table.len()?;
+        info!("migrate {name} ({len} rows)");
+        let ind = (len as usize / 1000) + 1;
+        for (i, entry) in old_table.iter()?.enumerate() {
+            let (key, values) = entry?;
+            let key = key.value();
+            if i > 0 && i % 1000 == 0 {
+                info!("    {name} {i:>ind$}/{len}");
```

Comment:
<img width="399" alt="Screenshot 2024-04-04 at 17 05 04" src="https://github.com/n0-computer/iroh/assets/5057878/e7f67c4a-5095-49b9-9f9e-164c07a3ad42">

Dunno how it is supposed to work, but it looks like this